pkg/controller/sparkapplication/submission.go

```diff
@@ -272,7 +272,7 @@ func addDriverConfOptions(app *v1beta1.SparkApplication, submissionID string) ([
 
 	if app.Spec.Driver.Cores != nil {
 		driverConfOptions = append(driverConfOptions,
-			fmt.Sprintf("spark.driver.cores=%f", *app.Spec.Driver.Cores))
+			fmt.Sprintf("spark.driver.cores=%d", int(*app.Spec.Driver.Cores)))
```

Comment:
this looks fine but I thought we tested with spark 3.0 and it worked fine with float ?

Reply:
Yeah for some reason it work in flyte...But spark 3 does not accept float...It may related to the fact that flyte sends the spark config as string...